examples/allennlp/allennlp_jsonnet.py

```diff
@@ -24,13 +24,16 @@
 import shutil
 
 import optuna
+from optuna.integration.allennlp import dump_best_config
 from optuna.integration import AllenNLPExecutor
 
 
 # This path trick is used since this example is also
 # run from the root of this repository by CI.
 EXAMPLE_DIR = os.path.dirname(os.path.abspath(__file__))
+CONFIG_PATH = os.path.join(EXAMPLE_DIR, "classifier.jsonnet")
 MODEL_DIR = "result"
+BEST_CONFIG_PATH = "best.classifier.json"
```

Comment:
Oh the output doesn't have to be a jsonnet (c.f. https://github.com/optuna/optuna/blob/master/tests/integration_tests/allennlp_tests/test_allennlp.py#L127)?

Reply:
Good question. 😅

In fact, all the variables defined in the jsonnet file are rewritten with the values found in the trial and embedded in the body, so the output file is valid as both jsonnet and json.
Do you thinks I should update [jsonnet](https://github.com/optuna/optuna/blob/master/tests/integration_tests/allennlp_tests/test_allennlp.py#L127) to be `result.json`?

Reply:
So the input and output file types are different? I'm not familiar with jsonnet but should we document that in [`dump_best_config`](https://github.com/optuna/optuna/blob/master/optuna/integration/allennlp.py#L24)? I'm fine with either if they are aligned.

Reply:
You're right. `dump_best_config` accepts both Json and Jsonnet files and dumps `config.json`, which is consistent with `allennlp train`. ([train](https://github.com/allenai/allennlp/blob/7d71398b44636c52e811467a684c3099c64009ec/allennlp/commands/train.py#L175-L238), [dump](https://github.com/allenai/allennlp/blob/7d71398b44636c52e811467a684c3099c64009ec/allennlp/models/archival.py#L84-L127))

I'll make another PR to modify the test and put a note that `dump_best_config` create a JSON file. I'm sorry for confusing you.

Reply:
Sounds good. And thanks for the explanation, it helps since I'm not familiar with how AllenNLP works.